nemo/collections/nlp/data/question_answering_squad/qa_dataset.py

```diff
@@ -136,6 +130,14 @@ def __init__(
                     with open(cached_features_file, "wb") as writer:
                         pickle.dump(self.features, writer)
 
+        # check number of samples. Should be either -1 not to limit or positive number
+        if num_samples == 0:
+            raise ValueError(
```

Comment:
Could you move this section up before line 110, then you can remove index checks down below

Reply:
wait... you mean: before use_cache?

Reply:
I think its best to move up line 134- 139 before line 110

Reply:
Corrected